application/src/main/java/org/ehrbase/application/abac/CustomMethodSecurityExpressionRoot.java

```diff
@@ -225,35 +231,77 @@ private boolean checkAbac(String type, String subject, Object payload, String co
 
     /**
      * Handles organization ID extraction. Uses token's claim.
-     * @param jwt Token
+     * @param token Token
      * @param requestMap ABAC request attribute map to add the result
      */
-    private void organizationHandling(JwtAuthenticationToken jwt, Map<String, Object> requestMap) {
-        if (jwt.getTokenAttributes().containsKey(abacConfig.getOrganizationClaim())) {
-            String orgaId = (String) jwt.getTokenAttributes().get(abacConfig.getOrganizationClaim());
-            requestMap.put(ORGANIZATION, orgaId);
-        } else {
-            // organization configured but claim not available
+    private void organizationHandling(AbstractAuthenticationToken token, Map<String, Object> requestMap) {
+        String orgId = getRequestedClaim(token, abacConfig.getOrganizationClaim());
+
+        if (isBlank(orgId)) {
+            // organization configured but claim not available or empty
             throw new IllegalArgumentException(
-                    "ABAC use of an organization claim is configured but " + "can't be retrieved from the given JWT.");
+                    "ABAC use of an organization claim is configured but can't be retrieved from the given JWT.");
+        }
+
+        requestMap.put(ORGANIZATION, orgId);
+    }
+
+    /**
+     * Extracts the requested claim from the token's claims.
+     * @param token Token
+     * @param requestedAbacClaim The claim to be retrieved
+     * @return The value of the requested claim
+     */
+    private String getRequestedClaim(AbstractAuthenticationToken token, String requestedAbacClaim) {
+        String claim = EMPTY;
+        Object principal = token.getCredentials();
+        if (principal instanceof Jwt jwt) {
+            Map<String, Object> claims = jwt.getClaims();
+            if (claims != null && claims.containsKey(requestedAbacClaim)) {
```

Comment:
Lines 260 and 265 are the same, move them into a Function<,>

Reply:
claims is different Map<String, Object> and Map<String, Claim>